StripePayments/StripePayments/Source/PaymentHandler/STPPaymentHandler.swift

```diff
@@ -1183,250 +1133,188 @@ public class STPPaymentHandler: NSObject {
                         intentID: currentAction.intentStripeID ?? ""
                     )
 
-                    if let authParams = authRequestParams,
-                        let transaction = transaction
-                    {
-                        currentAction.threeDS2Transaction = transaction
-                        currentAction.apiClient.authenticate3DS2(
-                            authParams,
-                            sourceIdentifier: useStripeSDK.threeDSSourceID ?? "",
-                            returnURL: currentAction.returnURLString,
-                            maxTimeout: currentAction.threeDSCustomizationSettings
-                                .authenticationTimeout,
-                            publishableKeyOverride: useStripeSDK.publishableKeyOverride
-                        ) { (authenticateResponse, error) in
-                            if let authenticateResponse = authenticateResponse,
-                                error == nil
-                            {
-
-                                if let aRes = authenticateResponse.authenticationResponse {
-
-                                    if aRes.isChallengeRequired {
-                                        let challengeParameters = STDSChallengeParameters(
-                                            authenticationResponse: aRes
-                                        )
+                    guard let authParams = authRequestParams, let transaction else {
+                        currentAction.complete(
+                            with: STPPaymentHandlerActionStatus.failed,
+                            error: self._error(
+                                for: .stripe3DS2ErrorCode,
+                                loggingSafeErrorMessage: transaction == nil ? "Missing transaction." : "Missing auth request params."
+                            )
+                        )
+                        return
+                    }
+                    currentAction.threeDS2Transaction = transaction
+                    currentAction.apiClient.authenticate3DS2(
+                        authParams,
+                        sourceIdentifier: useStripeSDK.threeDSSourceID ?? "",
+                        returnURL: currentAction.returnURLString,
+                        maxTimeout: currentAction.threeDSCustomizationSettings
+                            .authenticationTimeout,
+                        publishableKeyOverride: useStripeSDK.publishableKeyOverride
+                    ) { (authenticateResponse, error) in
+                        guard let authenticateResponse, error == nil else {
+                            let error = error ?? self._error(for: .stripe3DS2ErrorCode, loggingSafeErrorMessage: "Missing authenticate response")
```

Comment:
technically it is possible for us to enter this code if there is an `authenticationResponse` and there is an error, and we would log that we are missing an authentication response.  Maybe lets break this up a bit?

Also, might be more readable if we rename `authenticateResponse` to something like `result` or `authResult` or `authenticateResult`

Reply:
That can't actually happen, we'll only use "Missing authenticate" if the error is nil. But it's very hard to read, so I'll break it up anyways.

I don't love the 'authenticateResponse' name but why would `result` etc be better? The class itself is called `STP3DS2AuthenticateResponse' and represents the `/authenticate` endpoint response.  Unless you feel strongly I'd prefer to put renaming out of scope for this PR.